src/test/java/edu/wpi/grip/core/ConnectionTest.java

```diff
@@ -47,4 +51,26 @@ public void testInputSocketNotNull() {
         Socket<Double> foo = new Socket<>(eventBus, fooHint);
         Connection<Double> connection = new Connection<>(eventBus, foo, null);
     }
+
+    @Test
+    public void testForwardsConnection() {
+        Socket<Double> output = new Socket<>(eventBus, fooHint);
+        output.setDirection(Socket.Direction.OUTPUT);
+
+        Socket<Double> input= new Socket<>(eventBus, fooHint);
+        input.setDirection(Socket.Direction.INPUT);
+
+        new Connection<Double>(eventBus, output, input);
```

Comment:
What is this test proving? Some sort of assertion should be here even if you are just testing that an exception isn't thrown.